src/openforms/authentication/contrib/digid_oidc/models.py

```diff
@@ -0,0 +1,201 @@
+        ),
+        blank=True,
+    )
+
+    oidc_redirect_allowed_hosts = ArrayField(
+        verbose_name=_("Allowed redirect hosts"),
+        base_field=models.CharField(_("Allowed redirect hosts"), max_length=1000),
+        default=list,
+        blank=True,
+        help_text=_(
+            "Hosts that are allowed for redirection after authentication is successful"
+        ),
+    )
+
+    # Keycloak specific config
+    oidc_keycloak_idp_hint = models.CharField(
+        _("Keycloak Identity Provider hint"),
+        max_length=1000,
+        help_text=_(
+            "Specific for Keycloak: parameter that indicates which identity provider "
+            "should be used (therefore skipping the Keycloak login screen)."
+        ),
+        blank=True,
+    )
+
+    class Meta:
+        verbose_name = _("OpenID Connect configuration for DigiD")
+
+    def __str__(self):
+        return force_text(self._meta.verbose_name)
+
+    @property
+    def oidc_rp_scopes(self):
+        """
+        Scopes should be formatted as a string with spaces
+        """
+        return " ".join(self.oidc_rp_scopes_list)
+
+    @classmethod
+    def clear_cache(cls):
+        cache_name = getattr(
+            settings, "OIDC_CACHE", oidc_settings.MOZILLA_DJANGO_OIDC_DB_CACHE
+        )
+        if cache_name:
+            cache = get_cache(cache_name)
+            cache_key = cls.get_cache_key()
+            cache.delete(cache_key)
+
+    def set_to_cache(self):
+        cache_name = getattr(
+            settings,
+            "MOZILLA_DJANGO_OIDC_DB_CACHE",
+            oidc_settings.MOZILLA_DJANGO_OIDC_DB_CACHE,
+        )
+        if not cache_name:
+            return None
+        cache = get_cache(cache_name)
+        cache_key = self.get_cache_key()
+        timeout = getattr(
+            settings,
+            "MOZILLA_DJANGO_OIDC_DB_CACHE_TIMEOUT",
+            oidc_settings.MOZILLA_DJANGO_OIDC_DB_CACHE_TIMEOUT,
+        )
+        cache.set(cache_key, self, timeout)
+
+    @classmethod
+    def get_cache_key(cls):
+        prefix = getattr(
+            settings,
+            "MOZILLA_DJANGO_OIDC_DB_PREFIX",
+            CUSTOM_OIDC_DB_PREFIX,
+        )
+        return "%s:%s" % (prefix, cls.__name__.lower())
+
+    @classmethod
+    def get_solo(cls):
+        cache_name = getattr(
+            settings,
+            "MOZILLA_DJANGO_OIDC_DB_CACHE",
+            oidc_settings.MOZILLA_DJANGO_OIDC_DB_CACHE,
+        )
+        if not cache_name:
+            obj, created = cls.objects.get_or_create(pk=cls.singleton_instance_id)
+            return obj
+        cache = get_cache(cache_name)
+        cache_key = cls.get_cache_key()
+        obj = cache.get(cache_key)
+        if not obj:
+            obj, created = cls.objects.get_or_create(pk=cls.singleton_instance_id)
+            obj.set_to_cache()
+        return obj
```

Comment:
I'd like to see this repetition of code moved to a mozilla-django-oidc-db model mixin, and in general a critical look at what can be provided in an abstract base class (such as all the endpoint settings) to reduce repetition of code.

The prefix should then probably be a (dynamic) class attribute, django has a `django.utils.decorators.classproperty` decorator for that.

Reply:
I removed this code and allowed overriding of the prefix in `mozilla-django-oidc-db`, see https://github.com/maykinmedia/mozilla-django-oidc-db/pull/36